### PR TITLE
fix(resource manager): break "remove channels when disconnected" invariant from state machine

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-13599

Fixes https://github.com/emqx/emqx/issues/13979

Release version: v/e5.8.4

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [na] ~Added tests for the changes~ relying on vast array of existing bridge tests
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change

### CODE REVIEW

**Comment by zmstone** [comment]
fixes https://github.com/emqx/emqx/issues/13979 ?

File: apps/emqx_resource/src/emqx_resource_manager.erl [hunk 1]
```diff
@@ -1404,27 +1427,40 @@ resource_not_connected_channel_error_msg(ResourceStatus, ChannelId, Data1) ->
...
+            [NewChanStatus, PreviousChanStatus]
+        ),
+    Timeout = get_channel_health_check_interval(ChannelId, ConfigSources, Data),
     Event = #start_channel_health_check{channel_id = ChannelId},
     [generic_timeout_action(Event, Timeout, Event)].
 
-get_channel_health_check_interval(ChannelId, NewChanStatus, PreviousChanStatus, Data) ->
+-spec retry_add_channel_action(channel_id(), map(), data()) -> retry_add_channel_action().
+retry_add_channel_action(ChannelId, ChannelConfig, Data) ->
+    Timeout = get_channel_health_check_interval(ChannelId, [ChannelConfig], Data),
```
Comment (thalesmg) [inline apps/emqx_resource/src/emqx_resource_manager.erl:1449]: Q: maybe this should instead be `resume_interval`, if in `resource_opts`?  :thinking:

File: apps/emqx_resource/src/emqx_resource_manager.erl [hunk 2]
```diff
@@ -1340,33 +1362,34 @@ channels_health_check(?status_connecting = _ConnectorStatus, Data0) ->
         ChannelsWithNewAndPrevErrorStatuses
     ),
     Data1 = Data0#data{added_channels = NewChannels},
-    update_state(Data1);
-channels_health_check(ConnectorStatus, Data0) ->
-    %% Whenever the resource is not connected and not connecting:
+    {_Actions = [], update_state(Data1)};
+channels_health_check(?status_disconnected = ConnectorStatus, Data1) ->
+    %% Whenever the resource is disconnected:
     %% 1. Remove all added channels
```
Comment (zmstone) [inline apps/emqx_resource/src/emqx_resource_manager.erl:1368]: stale comment ?

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-5345)

**Pull request merged**